macros/edr/data_monitoring/monitors_query/dimension_monitoring_query.sql

```diff
@@ -17,75 +25,93 @@
             and edr_bucket_end <= {{ elementary.edr_cast_as_timestamp(max_bucket_end) }}
         ),
 
-        filtered_monitored_table as (
+        time_filtered_monitored_table as (
             select *,
-                   {{ concat_dimensions_sql_expression }} as dimension_value,
                    {{ elementary.get_start_bucket_in_data(timestamp_column, min_bucket_start, metric_properties.time_bucket) }} as start_bucket_in_data
-            from {{ monitored_table_relation }}
+            from filtered_monitored_table
             where
                 {{ elementary.edr_cast_as_timestamp(timestamp_column) }} >= (select min(edr_bucket_start) from buckets)
                 and {{ elementary.edr_cast_as_timestamp(timestamp_column) }} < (select max(edr_bucket_end) from buckets)
-            {% if metric_properties.where_expression %}
-                and {{ metric_properties.where_expression }}
-            {% endif %}
         ),
 
-        {# Outdated dimension values are dimensions with all metrics of 0 in the range of the test time #}
-        dimension_values_without_outdated as (
-            select distinct 
+        all_dimension_metrics as (
+            select
+                bucket_end,
                 dimension_value,
-                sum(metric_value)
+                metric_value,
+                row_number () over (partition by dimension_value order by bucket_end desc) as row_number
             from {{ ref('data_monitoring_metrics') }}
             where full_table_name = {{ full_table_name_str }}
                 and metric_name = {{ elementary.edr_quote(metric_name) }}
-                and dimension = {{ elementary.edr_quote(dimensions_string) }}
-                and {{ elementary.edr_cast_as_timestamp('bucket_end') }} >= {{ elementary.edr_cast_as_timestamp(min_bucket_start) }}
+                and metric_properties = {{ elementary.dict_to_quoted_json(metric_properties) }}
+        ),
+
+        training_set_dimensions as (
+            select distinct
+                dimension_value,
+                min(bucket_end) as dimension_min_bucket_end,
+                sum(metric_value)
+            from all_dimension_metrics
+            where row_number <= {{ test_configuration.min_training_set_size }}
             group by 1
+            {# Remove outdated dimension values (dimensions with all metrics of 0 in the range of the test time) #}
             having sum(metric_value) > 0
         ),
 
-        dimension_values_union as (
-            select distinct *
-            from (
-                select distinct 
-                    dimension_value,
-                    1 as joiner
-                from dimension_values_without_outdated
-                union all
-                select distinct
-                    dimension_value,
-                    1 as joiner
-                from filtered_monitored_table
-            ) results
+        unique_previous_dimension_values as (
```

Comment:
Could they be not unique?
On `training_set_dimensions` we already group by the dimension value and distinct it.
This table will be the same without the SUM and wit a joiner.
Maybe we could just add the joiner to the `training_set_dimensions` table?